src/run_autodiscovery.py

```diff
@@ -0,0 +1,237 @@
+import asyncio
+import json
+import os
+import re
+
+from dataclasses import dataclass
+from lib.clientsession_provider import init_gcp_client_session
+from lib.context import LoggingContext, MetricsContext, get_should_require_valid_certificate
+from lib.credentials import create_token, fetch_dynatrace_api_key, fetch_dynatrace_url
+from lib.metrics import Metric
+from typing import List, Text, Any, Dict
+from lib.configuration import config
+
+logging_context = LoggingContext(None)
+projectID="dynatrace-gcp-extension"
+discovered_resource_type="cloud_function"
+
+
+
+@dataclass(frozen=True)
+class MetricMetadata:
+    """Describes singular GCP service to ingest data from."""
```

Comment:
If it describes singular GCP service, why it is named MetricMetadata? 
It looks more like it describes single metric?